ssl/quic/quic_rx_depack.c

```diff
@@ -0,0 +1,612 @@
+                ok = 0;
+
+            OPENSSL_free(ack_ranges);
+            if (!ok)
+                return 0;
+            break;
+        }
+
+        case OSSL_QUIC_FRAME_TYPE_RESET_STREAM: {
+            OSSL_QUIC_FRAME_RESET_STREAM frame_data;
+
+            if (!ossl_quic_wire_decode_frame_reset_stream(pkt, &frame_data))
+                return 0;
+
+            switch (mode) {
+            case collect_ack_data:
+                /* This frame makes the packet ACK eliciting */
+                ackm_data->is_ack_eliciting = 1;
+                break;
+            case process_frames: {
+                QUIC_STREAM *stream
+                    = ssl_get_stream(connection, frame_data.stream_id);
+                int stream_type = ssl_get_stream_type(stream);
+
+                ssl_close_stream(stream);
+                if ((stream_type & SSL_STREAM_TYPE_W) != 0) {
+                    OSSL_STATM *statm = GET_CONN_STATM(connection);
+
+                    ssl_close_connection(connection);
+                    ossl_statm_set_error_state(statm, QUIC_STREAM_STATE_ERROR);
+
+                    /*
+                     * There's no point to continue with this packet.  However,
+                     * processing it was a success, so we return with a happy
+                     * face, and leave it to the state machine what to do with
+                     * the general STREAM_STATE_ERROR state.
+                     */
+                    return 1;
+                }
+                break;
+            }
+            }
+            break;
+        }
+        case OSSL_QUIC_FRAME_TYPE_STOP_SENDING: {
+            OSSL_QUIC_FRAME_STOP_SENDING frame_data;
+
+            if (!ossl_quic_wire_decode_frame_stop_sending(pkt, &frame_data))
+                return 0;
+
+            switch (mode) {
+            case collect_ack_data:
+                /* This frame makes the packet ACK eliciting */
+                ackm_data->is_ack_eliciting = 1;
+                break;
+            case process_frames: {
+                QUIC_STREAM *stream
+                    = ssl_get_stream(connection, frame_data.stream_id);
+                int stream_type = ssl_get_stream_type(stream);
+
+                ssl_close_stream(stream);
+                if ((stream_type & SSL_STREAM_TYPE_R) != 0) {
+                    OSSL_STATM *statm = GET_CONN_STATM(connection);
+
+                    ssl_close_connection(connection);
+                    ossl_statm_set_error_state(statm, QUIC_STREAM_STATE_ERROR);
+
+                    /*
+                     * There's no point to continue with this packet.  However,
+                     * processing it was a success, so we return with a happy
+                     * face, and leave it to the state machine what to do with
+                     * the general STREAM_STATE_ERROR state.
+                     */
+                    return 1;
+                }
+                break;
+            }
+            }
```

Comment:
This looks very wrong from a style perspective. But I'm not sure in our coding style how we are supposed to do this.

Reply:
This is a consequence of treating `case` lines like labels, i.e. outdent them:

``` C
switch (expr) {
case foo:
    something();
    break;
case bar: {
    int local_num = 0;

    somethingelse(&local_num);
    break;
}
}
```

The other option would be to not outdent the `case` lines:

``` C
switch (expr) {
    case foo:
        something();
        break;
    case bar: {
        int local_num = 0;

        somethingelse(&local_num);
        break;
    }
}
```

In this regard, we're actually a bit undecided (I've seen code where the `case` lines aren't outdented)...  however, the outdented layout was part of the original source reformatting, and is also the current setup in https://github.com/openssl/openssl/blob/master/doc/openssl-c-indent.el

Reply:
> we're actually a bit undecided

The coding style is very clear that you are supposed to line up the switch and case statements (as you have done). See chapter 3:

https://www.openssl.org/policies/technical/coding-style.html#chapter-3-placing-braces-and-spaces

But it doesn't really discuss the instance where you have a block inside `{}` as part of that case. This is actually something I don't see much (I don't know if we have other instances of this in the code). I've often just avoided the issue by not using `{}` and just declare any variables in the next block out.

Reply:
Perhaps the answer is this:

```` C
switch (expr) {
case foo:
    something();
    break;
case bar: {
        int local_num = 0;

        somethingelse(&local_num);
        break;
    }
}
````
This actually makes sense. Opening a new block with `{` should add an extra level of indent to where you would otherwise be without the new block.

Reply:
> (I don't know if we have other instances of this in the code).

To no one's surprise, the answer is a grep away 😉 : 

``` console
$ git grep -E 'case .*: *\{'
crypto/context.c:    case OSSL_LIB_CTX_RAND_CRNGT_INDEX: {
crypto/ec/ecx_meth.c:    case ASN1_PKEY_CTRL_SET1_TLS_ENCPT: {
crypto/evp/evp_enc.c:    case EVP_CTRL_TLS1_1_MULTIBLOCK_AAD: {
crypto/evp/evp_enc.c:    case EVP_CTRL_TLS1_1_MULTIBLOCK_ENCRYPT: {
providers/fips/self_test_kats.c:        case OSSL_PARAM_UNSIGNED_INTEGER: {
providers/fips/self_test_kats.c:        case OSSL_PARAM_UTF8_STRING: {
providers/fips/self_test_kats.c:        case OSSL_PARAM_OCTET_STRING: {
providers/fips/self_test_kats.c:        case OSSL_PARAM_INTEGER: {
```

> I've often just avoided the issue by not using `{}` and just declare any variables in the next block out.

I strongly dislike having to place such fairly block local variables a longer distance away.  It makes code harder to read.

> Perhaps the answer is this:
> 
> ```c
> switch (expr) {
> case foo:
>     something();
>     break;
> case bar: {
>         int local_num = 0;
> 
>         somethingelse(&local_num);
>         break;
>     }
> }
> ```

No, please no.  Ugly!

Reply:
I could, of course, add an unnecessary `default:` case, like, for example, here:

https://github.com/openssl/openssl/blob/2f1112b22a826dc8854b41b60a422c987f8ddafb/providers/fips/self_test_kats.c#L181-L188

(without it, you would have seen the same `}` "problem" as you see here)

Reply:
Doesn't look ugly to me. Looks quite reasonable IMO and solves the *really* ugly formatting we currently have in this PR.

Reply:
> I could, of course, add an unnecessary default: case, like, for example, here:

It doesn't really solve the problem IMO. That "trick" wouldn't work if you needed a block in your default case too.

As @t8m says, this extra indenting (as per my proposal) is exactly what you would do if you put a block in the middle of any other code, e.g.:

```` C
switch (expr) {
case foo:
    something();
    break;
case bar:
    thing();
    {
        int local_num = 0;

        somethingelse(&local_num);
    }
    break;
}
````
So if the above is ok then surely this is too:
```` C
switch (expr) {
case foo:
    something();
    break;
case bar:
    {
        int local_num = 0;

        thing();
        somethingelse(&local_num);
        break;
    }
}
````
And then putting the opening `{` on the same line as the case is just a logical next step.

Reply:
Yeah, disagreement is possible, but the current way is IMO outright confusing.

Reply:
We might want to formalize this in the coding style guide.  And codify it in [`doc/openssl-c-indent.el`](https://github.com/openssl/openssl/blob/master/doc/openssl-c-indent.el)...